src/resources/effects/lua.rs

```diff
@@ -0,0 +1,134 @@
+use crate::resources::color::Color;
+use jsonschema::JSONSchema;
+use mlua::{Error, Function, Lua, LuaSerdeExt, Table, Value};
+use std::fs;
+
+#[derive(Debug)]
+pub enum InvalidEffectError {
+    MissingSchema,
+    InvalidSchema,
+}
+
+#[derive(Debug)]
+pub enum LuaEffectLoadError {
+    File(std::io::Error),
+    Lua(Error),
+    Effect(InvalidEffectError),
+}
+
+#[derive(Debug)]
+pub enum LuaEffectRuntimeError {
+    Lua(Error),
+    WrongColorsLen,
+    MissingTickFunction,
+    MissingFrameworkImport,
+}
+
+#[allow(unused)]
```

Comment:
Do we still need this?

Reply:
Ouais a cause des json schema
Va falloir check comment on veut manage ca